enhancements/metadata/user-tag-management.md

```diff
@@ -0,0 +1,372 @@
+7. Retry mechanism for failed operation during reconciliation.
+8. Provide metrics collection and audit logging features for controller and user operations.
+9. UI frontend for managing tags.
+
+## Proposal 
+
+A new kubernetes controller to generates actions based on tag specification defined by a custom resource. The controller
+reads existing data, applies tags based on opcode and maximum limit policy. The controller would require an initial tag or label, 
+that can be used to identify cloud resources to apply and manage tags. Cloud resources can be ignored from being managed 
+by the controller by removing the classifier tag on the cloud resource. Controller reconciles the tag list on 
+cloud resources periodically at intervals configured by sync period for the controller. 
+
+### Controller configuration
+
+#### Maximum limit policy
+
+Maximum limit policy indicates whether the input tag list can be applied partially when the length of the list exceeds the 
+allowed limit on the cloud resource. The result might differ when applied along overwrite policy. Following are the list of valid values.
+
+1. POLICY_APPLY_PARTIAL - Allows input tag list applied to be partial on cloud resource. 
+2. POLICY_APPLY_STRICT - Applies input tag list strictly in complete.
+
+### Opcodes
+
+Controller supports different opcodes for actions to add, update and delete tags. Following are the supported opcodes
+1. "add" - Controller adds tags to the cloud resources. In case of an existing entry for the tag, controller reports failure.
+2. "update" - Controller updates existing tags for the cloud resources. If there is no existing entry, controller reports failure.
+3. "delete" - Controller removes tags from the cloud resources. In case of specific errors, user intervention may be required to remove tags manually
+from the cloud resources.
+
+### High-level workflow
+
+#### Bootstrap
+1. User starts the controller with cloud service provider credentials and sync period set.
+2. Controller starts to list/watch custom resource.
+3. User creates `AWSMetadata` object with tag list details.
+4. User creates `CloudMetadata` configuration with controller configuration.
+5. Based on the configuration, controller queries and lists all resources based on classifier tag.
+6. Creates list of existing tags on cloud resource.
+7. Add/update tag workflow is initiated.
+
+#### Add/Update tags
+
+1. User creates `AWSMetadata` object with tag list details.
+2. User creates `CloudMetadata` configuration with controller configuration.
+3. Controller updates ready condition to false in `MetadataStatus.status`.
+4. Controller validates tag entries from `AWSMetadata.spec.resourcetags` and populates a new list of tags to be added to cloud resource.
+5. Controller trims the tag list as per maximum limit policy specific to cloud resource.
+6. Controller performs lexicographic sorting.
+7. Controller replaces the tag list on the cloud resource.
```

Comment:
this says replaces, but later you say it adds them to the existing tags.  I think you actually intend to do the latter, so this step should be rewritten.